src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/JsonHubProtocol.cs

```diff
@@ -344,7 +341,7 @@ private HubMessage ParseMessage(Utf8BufferTextReader textReader, IInvocationBind
 
         private void WriteMessageCore(HubMessage message, Stream stream)
         {
-            using (var writer = new JsonTextWriter(new StreamWriter(stream, _utf8NoBom, 1024, leaveOpen: true)))
+            using (var writer = JsonUtils.CreateJsonTextWriter(new StreamWriter(stream, _utf8NoBom, 1024, leaveOpen: true)))
```

Comment:
Who is closing the stream writer here?

Reply:
Does it need to be closed?

As far as I know Close just calls Flush and closes the underlying stream, but we're explicitly calling Flush and the underlying stream is a memory stream that doesn't need closing.

Reply:
nope carry on, it does nothing if leave open is true